lib/origen_testers/igxl_based_tester/base/edgesets.rb

```diff
@@ -0,0 +1,53 @@
+module OrigenTesters
+  module IGXLBasedTester
+    class Base
+      class Edgesets
+        include ::OrigenTesters::Generator
+
+        attr_accessor :es
+
+        OUTPUT_PREFIX = 'ES_'
+        OUTPUT_POSTFIX = ''
+
+        def initialize # :nodoc:
+          @es = {}
+        end
+
+        def add(esname, pin, edge, options = {})
+          esname = esname.to_sym unless esname.is_a? Symbol
+          pin = pin.to_sym unless pin.is_a? Symbol
+          @es.key?(esname) ? @es[esname].add_edge(pin, edge) : @es[esname] = platform::Edgeset.new(esname, pin, edge, options)
+          @es
+        end
+
+        def finalize(options = {})
+        end
+
+        # Equality check to compare full contents of edge object
+        def edges_eql?(edge1, edge2)
+          edge1.d_src == edge2.d_src && edge1.d_fmt == edge2.d_fmt &&
+            edge1.d0_edge == edge2.d0_edge && edge1.d1_edge == edge2.d1_edge &&
+            edge1.d2_edge == edge2.d2_edge && edge1.d3_edge == edge2.d3_edge &&
+            edge1.c_mode == edge2.c_mode &&
+            edge1.c1_edge == edge2.c1_edge && edge1.c2_edge == edge2.c2_edge &&
+            edge1.t_res == edge2.t_res && edge1.clk_per == edge2.clk_per
+        end
```

Comment:
@ginty Unfortunately, I found your statement not to be true regarding the == method as defined in the Object class.  Here is a web snippet I found that supports my finding:
string1 = "abc"
class MyObject
end
object1 = MyObject.new
object2 = object1
object3 = MyObject.new

puts "Object 1 is == to object 2: #{object1 == object2}"
puts "Object 1 is eql? to object 2: #{object1.eql? object2}"
puts "Object 1 is equal? to object 2: #{object1.equal? object2}"
puts "Object 1 is == to object 3: #{object1 == object3}"
puts "Object 1 is eql? to object 3: #{object1.eql? object3}"
The output is:
Object 1 is == to object 2: true
Object 1 is eql? to object 2: true
Object 1 is equal? to object 2: true
Object 1 is == to object 3: false
Object 1 is eql? to object 3: false

I did, however, find an alternative to avoid the ugliness I had before:
edge1.instance_values == edge2.instance_values


From: Stephen McGinty [mailto:notifications@github.com]
Sent: Thursday, March 17, 2016 12:02 PM
To: Origen-SDK/origen_testers <origen_testers@noreply.github.com>
Cc: Aaron Burgmeier <aaron.burgmeier@nxp.com>
Subject: Re: [origen_testers] Support for Edgeset/Timeset sheet generation (#8)


In lib/origen_testers/igxl_based_tester/base/edgesets.rb<https://github.com/Origen-SDK/origen_testers/pull/8#discussion_r56539603>:

> +          @es.key?(esname) ? @es[esname].add_edge(pin, edge) : @es[esname] = platform::Edgeset.new(esname, pin, edge, options)

> +          @es

> +        end

> +

> +        def finalize(options = {})

> +        end

> +

> +        # Equality check to compare full contents of edge object

> +        def edges_eql?(edge1, edge2)

> +          edge1.d_src == edge2.d_src && edge1.d_fmt == edge2.d_fmt &&

> +            edge1.d0_edge == edge2.d0_edge && edge1.d1_edge == edge2.d1_edge &&

> +            edge1.d2_edge == edge2.d2_edge && edge1.d3_edge == edge2.d3_edge &&

> +            edge1.c_mode == edge2.c_mode &&

> +            edge1.c1_edge == edge2.c1_edge && edge1.c2_edge == edge2.c2_edge &&

> +            edge1.t_res == edge2.t_res && edge1.clk_per == edge2.clk_per

> +        end

@aaronburg<https://github.com/aaronburg>, if you implement the == method it won't care about object ID unless you include it in your comparison. In fact the whole point of doing that is precisely because you still want to be able to consider two different objects equal if the attributes you are about are the same.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Origen-SDK/origen_testers/pull/8/files/8ffdad474d2cf9147fcfe3b93a662d2cc1d1461e#r56539603>